awx/api/views/__init__.py

```diff
@@ -3762,6 +3763,10 @@ def create(self, request, *args, **kwargs):
                 data = dict(passwords_needed_to_start=needed)
                 return Response(data, status=status.HTTP_400_BAD_REQUEST)
 
+        # Convert OrderedDict to JSON string
+        if request.data.get('extra_vars'):
+            request.data['extra_vars'] = json.dumps(request.data['extra_vars'])
+
```

Comment:
I didn't really want to change API behavior with this, since that is likely to disrupt the UI and other clients. You can do something similar in the awx.awx.ad_hoc_command module itself, and that would be the easiest solution the problem.

Reply:
I've moved the change to the ad_hoc_command module